client/components/PuzzleNavBar/PuzzleNavBar.jsx

```diff
@@ -26,6 +51,7 @@ export function PuzzleNavBar({puzzleNum, puzzleDesc}) {
               <Nav.Link href="/UserProfile">User Profile</Nav.Link>
               <Nav.Link onClick={handleShowDesc}>Puzzle Description</Nav.Link>
             </Nav>
+            <p className="timer">Time: {minutes}:{seconds?.toString().padStart(2, '0') || '00'}:{milliseconds?.toString().padStart(2, '0') || '00'}</p>
```

Comment:
Probably a little excessive to be showing milliseconds here, but this is more of a nitpick.

Reply:
..didn't you tell me to add milliseconds?

Reply:
It makes sense to have it in the final popup when you finish the puzzle, but it feels a little distracting to me to be in the PuzzleNavBar specifically. Again, this is a bit of a nitpick and it's fine if you don't want to change it